avatica/src/main/java/org/apache/calcite/avatica/AvaticaStatement.java

```diff
@@ -106,14 +106,44 @@ protected void executeInternal(String sql) throws SQLException {
     try {
       // In JDBC, maxRowCount = 0 means no limit; in prepare it means LIMIT 0
       final long maxRowCount1 = maxRowCount <= 0 ? -1 : maxRowCount;
-      Meta.ExecuteResult x =
-          connection.prepareAndExecuteInternal(this, sql, maxRowCount1);
+      while (true) {
+        // TODO limit this?
```

Comment:
I would like to have some limit here to prevent stuck threads for unknown reason. I think it is better to fail a single request than block an application thread forever.
If you do not like hard-coding "5", you might use "system property".

Otherwise, `while(true)` loops are prone to errors like 'never ending request'.